src/Integrations/Integrations/Symfony/SymfonyIntegration.php

```diff
@@ -82,6 +82,77 @@ public function init()
             ]
         );
 
+        \DDTrace\hook_method(
+            'Doctrine\ORM\UnitOfWork',
+            'executeInserts',
+            function ($This, $scope, $args) {
+                  $metadataClass = 'Doctrine\ORM\Mapping\ClassMetadata';
+                  if (
+                       !$args ||
+                       !isset($args[0]) ||
+                       !$args[0] ||
+                       !($args[0] instanceof $metadataClass)
+                  ) {
+                       return;
+                  }
+
+                  $entities = $This->getScheduledEntityInsertions();
+                  $userInterface = 'Symfony\Component\Security\Core\User\UserInterface';
+                  $found = 0;
+                  $userEntity = null;
+                  foreach ($entities as $entity) {
+                    if (! ($entity instanceof $userInterface)) {
+                        continue;
+                    }
+                    $found++;
+                    $userEntity = $entity;
+                  }
+
+                  if ($found != 1) {
+                    return;
+                  }
+
+                  \datadog\appsec\track_user_signup_event($userEntity->getUsername(), [], true);
```

Comment:
Just to be sure, so here we are going through **all** the entity insertions, and only tracking the signup event on **the last one**? Can't we directly grab the last element of the `entityInsertions` array? Maybe there is something I understood wrong there 🤔

Reply:
What this is trying to achieve is to trigger `track_user_signup_event` event when there is a sign up. On the loop I need to go through all the entities and check if there is one and only one user entity.  If there are multiple, it probably is a mass user creation process or something else but not a sign up